packages/easy-hydrator/src/ClassConstructorValuesResolver.php

```diff
@@ -5,26 +5,26 @@
 use ReflectionClass;
 use ReflectionMethod;
 use Symplify\EasyHydrator\Exception\MissingConstructorException;
-use Symplify\PackageBuilder\Strings\StringFormatConverter;
+use Symplify\EasyHydrator\ParameterValueGetter\ParameterValueGetterInterface;
```

Comment:
Why use an interface there? Seems like only 1 exact service is used, so interface can be removed

Reply:
Thank you for your question. I saw some importance in keeping it open for replacing with own implementation.

For example we want to throw exception when parameter is missing in body, instead of returning null. Not giving it interface and having it as final class would prevent doing it.

Reply:
That should be handled by configuration/collector then.

Reply:
My bad, completely missed that part. So interface is okay but need to put it into configuration (https://symfony.com/doc/current/bundles/configuration.html), is that right please?

Reply:
This one might be tricky to figure out. Depends what is the most preffered way to use this package.

Symfony config is over engeneered. Having parameter is better.

It still feels wrong to use parameter for functionality. Maybe having 2 methods in hydrator, one with exception throwing and one with silents pass would be better.

Like repository pattern has:

```php
// optionally returns null
$this->findBy(...);

// must return object or throw an exception
$this->getBy(...);
```

Reply:
Hmm, that could work! I will try to implement it this way.

Reply:
Unfortunate situation is that it would have to go through 3 levels:

Could add 2 methods to the highest level (public api i'd say) to throw exception on missing parameter in data:
- `strictlyHydrateArray()`
- `strictlyHydrateArrays()`

And internally passing a `$throw = true` parameter as it is the most straightforward and readable variant without much code duplication: 

Before:
```
Symplify\EasyHydrator\ClassConstructorValuesResolver::resolve(string $class, array $data): array
```

After:
```
Symplify\EasyHydrator\ClassConstructorValuesResolver::resolve(string $class, array $data, bool $throwExceptionOnMissingData = false): array
```

Are you okay with this?

Reply:
The complexity feels wrong here. I guess I still don't understand what's the reason to have 2 methods to hydrate objects. 

The value object code should be responsible for this. 

- When variable is nullable, allow null.
- When it's not nullable and is missing, throw an exception.

Could you maybe show me some 2 code snippets that shows it nicely?

Reply:
Also, calling might be faster. This might take hours :D